doc/md/fundamentals/2-types/10-options.md

```diff
@@ -0,0 +1,227 @@
+
+An option is defined using `?` followed by the type of the value it can hold.
+
+```motoko
+var username : ?Text = null;
+```
+
+`username` is an optional [`Text`](https://internetcomputer.org/docs/motoko/base/Text) value that starts as `null` (no username set).
+
+:::info Null semantics
+
+The constant `null` is the only value of the trivial `Null` type.  It also indicates an absent value in an optional type (`?T`). This makes `null` similar to `None` in languages like Rust, Scala, and `Nothing` in Haskell, and Motoko's option `?T` type similar to `Option<T>` in Rust, `Option[T]` in Scala, and `Maybe T` in Haskell.  In all these languages the type system enforces explicit handling of missing values by using a dedicated type to indicate optionality
+
+:::
+
+When a Motoko value has type `?T`, then is either `null` or some value, written `?value` (note leading `?`). The fundamental way to access the value is to use a `switch` expression, that explicitly handles both cases:
+
+``` motoko
+func displayName(option : ?Text) : Text {
+  switch option {
+     case (?user) { user };
+     case (null) { "Guest" };
+  }
+};
+displayName(null);
+```
+
+Sometimes, the verbosity of `switch` expression can make code harder to read, so Motoko also offers additional ways to handle option types.
+
+### Checking for presence
+
+To determine if an option contains a value, `Option.isSome` returns `true` if it is not `null`.
+
+```motoko
+import Option "mo:base/Option";
+import Debug "mo:base/Debug";
+
+let value : ?Nat = ?5;
+if (Option.isSome(value)) {
+    Debug.print("Value is present.");
+}
+```
+
+By leveraging the `Option` module, handling optional values becomes more concise and expressive, reducing the need for explicit [`switch`](https://internetcomputer.org/docs/motoko/fundamentals/control-flow/switch) statements.
+
+### Providing default values
+
+Instead of manually handling `null` cases with [pattern matching](https://internetcomputer.org/docs/motoko/fundamentals/pattern-matching), `Option.get` allows for cleaner fallback logic to ensure that missing values are safely replaced with a default.
+
+```motoko
+import Option "mo:base/Option";
+
+let username : ?Text = null;
+Option.get(username, "Guest"); // "Guest" if username is null
+```
+
+### Using options for error handling
+
+Options can be used to catch expected failures instead of calling a [`trap`](https://internetcomputer.org/docs/motoko/fundamentals/basic-syntax/traps), making a function return `null` when it encounters an invalid input.
+
+```motoko
+func safeDivide(a : Int, b : Int) : ?Int {
+    if (b == 0) null else ?(a / b);
+};
+
+let result1 = safeDivide(10, 2); // ?5
+safeDivide(10, 0); // null
+```
+
+### Let / else
+
+Another way to extract values from options, is use the `let ... else` pattern. This may be preferable over a `switch` expression, but only applies when the `else` expression can divert control elsewhere when the value fails to match the pattern used in the `let`
```

Comment:
```suggestion
Another way to extract values from options, is use the `let ... else` pattern. This may be preferable over a `switch` expression, but only applies when the `else` expression can divert control elsewhere when the value fails to match the pattern used in the `let`.
```